core/src/main/java/org/mule/runtime/core/routing/UntilSuccessful.java

```diff
@@ -267,6 +267,7 @@ public Processor getDlqMP() {
   public Processor getRoute() {
     final DefaultMessageProcessorChain chain = DefaultMessageProcessorChain.from(muleContext, routes.get(0));
     chain.setTemplateMuleContext(muleContext);
+    chain.setFlowConstruct(flowConstruct);
```

Comment:
Is there no way to initialize this route when object is constructed and then implemented `getOwnedMessageProcessors()` for lifecycle?